File: SortingAlgos/selection_sort.cpp
#include<iostream>
#include<vector>
using namespace std;

//selection sort -> first part of the array is sorted and other is not sorted

int getMinIndex(vector<int>& arr, int i) {
    int minIdx = i;
    for(int j=i; j<arr.size(); j++) { //search for minElement in the unsorted array
        if(arr[j] < arr[minIdx]) minIdx = j;
    }
    return minIdx; //return index of minElement in the unsorted part of the array
}

void selectionSort(vector<int> &arr) {
    for(int i=0; i<arr.size(); i++) {
        int minIdx = getMinIndex(arr,i);
        swap(arr[i],arr[minIdx]);
    }
}

int main() {
    int n;
    cin>>n;

    vector<int> arr;
    while(n--) {
        int x;
        cin>>x;
        arr.push_back(x);
    }

    selectionSort(arr);

    for(int i=0; i<arr.size(); i++) {
        cout<<arr[i]<<" ";
    }

    return 0;
}

File: SortingAlgos/selection_sort.js
function getMinIndex(arr,i) {
    let minIdx = i;
    for(let j=i+1; j<arr.length; j++) {
        if(arr[j] < arr[minIdx]) minIdx = j;
    }
    return minIdx;
}

function selectionSort(arr) {
    for(let i=0; i<arr.length; i++) {
        let minIdx = getMinIndex(arr,i);
        [arr[minIdx], arr[i]] = [arr[i], arr[minIdx]]; //swaping
    }
}

// let arr = [3,-1,6,-2,8];
let arr = [6,-3,1,-2,4];
// 6,-3,1,-2,4
selectionSort(arr);

console.log(arr);
